### PR TITLE
inspector_redesign

### PR DESCRIPTION
**Pull request by vpereverzev**
Ta-daaa!

As you know, we redesigned the inspectors for several months in a row. This PR contains the main functional part. Of course, we still need to fix a lot of minor problems and refine a couple of forgotten screens. But what we have now is very impressive. I really-really hope that you guys will also appreciate the results of sleepless nights.

Special thanks to:
- @anatoly-os - for help with the "Section breaks", "Spacers", "Clefs", "Brackets", "Articulations" and all kind of frames.
- @Howard-C  - thanks for all kinds of help with the input validation (“DoubleInputValidator” and “IntInputValidator”), as well as for many small(but important!) fixes.
- @igorkorsukov - for the intermediate reviews of my huge PR, from which the GITHUB burst at the seams.
- @Tantacrul - for the 24/7 consultation and awesome design mockups and conceptual vision.

Example screen:
изображение

This is not the last PR concerning the new inspector, but the main one :)

### CODE REVIEW

**Comment by LayneTheEpic** [comment]
That looks beautiful! :D

**Comment by Jojo-Schmitz** [comment]
Quite a lot of merge conflicts, so a rebase is definitly needed.

**Comment by Marr11317** [comment]
I just tested the new design.
Here's some thoughts about it:

 1) Where is the individual reset button? Where is the set as style? How do you know which values have been modified?

 2) When something is tabbed with the inspector, the latter doesn't have a name: 
image

3) This design looks similar to the one of tabs. Is it a tab? 
image 
image
Why not keep the same as palettes? ( a simple "palette" in the top right)
image

4) Checkboxes react weirdly to hover events: it looks like they get greyed out on hover, which is counterintuitive:

(it is even weirder in light mode)

5) in dark mode, some buttons look like they aren't buttons until they are hovered. Look at the difference between these two:
image image

6) In the following image, I think the greyed out text serves no purpose. Actually, the wide blue line below the active tab should be enough to tell him what he presently is looking at. Take it that way. What would you do if you wanted to actually disable the tab? You wouldn't be able to do anything more except completely hide it, which is a bad idea.
image

2 solutions (I actually think both should be applied):
 - keep the inactive tab's text dark (do not grey it)
 - accept hover events to show that this text is actually clickable.

7) hover events for the buttons inside spin boxes please.

8) Reset to default box goes out of screen. ( that might be my screen's problem, needing some adjustments, but a little margin would still be appreciated if possible. Anyway, why is this much widtch needed? )
image


9) (same image as above) The 3 points inside the icon are almost invisible in the selected blue.

10) Some text in comboboxes stay black even in dark them. Low contrast:
image

11) Now an extra click is needed to modify any specific property of an element? All the properties don't already appear when you select an element: you fist have to open the specific tab. That sounds like a huge waste of time when doing minor edits to a lot of elements.

12) (maybe out of the scope of this PR) some (many) old inspectors had tooltips indicating the precise behaviour of every complex actions. These tooltips completely disappeared. Now a lot of information is missing...

13) Some spinboxes stay in light mode even when dark mode is enabled. Have a look at the ambitus' inspector. 
image

The spinboxes don't look like ones.

14) Spinboxes don't react to up and down arrows for navigation. (they actually don't react to much)

15) ambitus inspector: Try setting the line thickness to anything (even 0). It won't let you go under 0.1, which will result in something like this: 
image

16) why is this the smallest possible inspector? I'll soone need 3 screens:
image


I actually don't want to dig up for more, since evey time I open an inspector, there's something (some of these are minor glitches, like the break inspector showing an empty "Notation" section, some are really not great to see)

image


Anyway. A lot of work has been put into this, and it clearly is cleaner in a UI point of view. The UX still needs some great love.

Most of these are small issues, but they add up and make the whole thing worse than it used to be (in my point of view). I still am really curious about the final output of this design, and really hope it will make a great enhancement to Musescore's current inspector.

Congrats for all the job done, let's finish that well.

**Comment by Marr11317** [comment]
Missing "o" at the end of tempo:
image

**Comment by Marr11317** [comment]
Weird behaviour of hover events:

**Comment by Howard-C** [comment]
@Marr11317 I discovered No.15 too, and a fix has already been merged to 3.x (and master later as well, hopefully). The problem was with `libmscore` code, not the inspector. See #6122.

**Comment by Howard-C** [comment]
> Weird behaviour of hover events:
> 

I still don't see what's wrong?

**Comment by Howard-C** [comment]
@Marr11317 For No.3, Martin told me that the big "Inspector" title was intended for MuseScore 4. I suppose the title for palettes will be changed.

**Comment by Howard-C** [comment]
@Marr11317 For No.11: It will indeed not be as convenient as it is in 3.x, but the tab you opened stays unfolded as long as you have one or more of the type of element selected.

**Comment by Tantacrul** [comment]
@Marr11317

All the things you mentioned have been accounted for in the design and are scheduled for development.

In addition there are a few other things missing. Mainly: the default state of the inspector when nothing is selected; A non-pop-up version of element settings when you select a single item & a system for setting global styles.

Also, there hasn’t been a visual polish pass yet. We need to do a quick bit of polish now for things like dark mode (as well as quite a few individual improvements for light mode). Then there is a more comprehensive UI pass planned later for the entire interface for MS4.

File: mtest/mscore/workspaces/tst_workspaces.cpp [hunk 1]
```diff
@@ -206,6 +206,7 @@ void TestWorkspaces::prepareStandardWorkspacesTestData()
 
 void TestWorkspaces::testEditPalette()
 {
+    QSKIP("Temporarily disabled tests for the time of refactoring commands in MS4");
```
Comment (Jojo-Schmitz) [inline mtest/mscore/workspaces/tst_workspaces.cpp:209]: If you can't beat'em, kill'em ;-)

Comment (vpereverzev) [reply mtest/mscore/workspaces/tst_workspaces.cpp:209]: why am I so unlucky
Снимок экрана от 2020-06-02 22-23-26

Comment (Jojo-Schmitz) [reply mtest/mscore/workspaces/tst_workspaces.cpp:209]: bad karma?

**Comment by anatoly-os** [comment]
FINALLY!

**Comment by LayneTheEpic** [comment]
I can't wait! :D

**Comment by anatoly-os** [comment]
It's better to use Inspector in Light theme since Dark theme is not fully supported, yet.

**Pull request merged**